asciigraf/asciigraf.py

```diff
@@ -15,18 +15,30 @@ def graph_from_ascii(network_string):
         "|":  [Point(0, -1), Point(0, 1)]
     }
     EDGE_CHARS = {"\\", "-", "/", "|"}
-    nodes = OrderedDict(node_iter(network_string))
-
-    node_chars = {}
-    for node_label, pos in nodes.items():
-        for offset, char in enumerate(node_label):
-            node_chars[pos + Point(offset, 0)] = node_label
+    nodes = list(node_iter(network_string))
+
+    node_chars = OrderedDict()
+    line_labels = {}
+    line_label_char_positions = set()
+    for node_label, root_position in nodes:
+        node_label_char_positions = {
+            root_position + Point(offset, 0)
+            for offset, _ in enumerate(node_label)
+        }
+        if node_label.startswith("(") and node_label.endswith(")"):
+            label_value = node_label[1:-1]
```

Comment:
was pleasantly surprised to find that this works even if `node_label == '()'` (at least on python 3.6)

Reply:
Yeah, you should just get an empty string on the edge attribute.

Reply:
Also 2.x

```
Python 2.7.10 (default, Feb  7 2017, 00:08:15)
[GCC 4.2.1 Compatible Apple LLVM 8.0.0 (clang-800.0.34)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> "()"[1:-1]
''
```

Reply:
It makes sense it's the substring with the same starting and ending index in the string.